packages/resource-service/src/request.js

```diff
@@ -0,0 +1,28 @@
+import ConnectionManager, {
+  EventTypes,
+  XHRConnection
+} from "connection-manager";
+import Rx from "rxjs";
+
+export default function request(url, options) {
+  return Rx.Observable.create(function(observer) {
+    const connection = new XHRConnection(url, options);
+
+    connection.addListener(EventTypes.PROGRESS, function(target) {
+      observer.next(target.response);
+    });
+    connection.addListener(EventTypes.ERROR, function(error) {
+      observer.error(error);
+    });
+    connection.addListener(EventTypes.CLOSE, function() {
+      observer.complete();
```

Comment:
🤔  wonder if we need to push another event before completing, to make sure we get the _complete_ response...